src/Common/tests/System/Net/Http/LoopbackProxyServer.cs

```diff
@@ -218,39 +239,67 @@ private async Task ProcessConnectMethod(NetworkStream clientStream, string remot
             {
                 try
                 {
-                    byte[] buffer = new byte[8000];
-                    int bytesRead;
-                    while ((bytesRead = await clientStream.ReadAsync(buffer, 0, buffer.Length).ConfigureAwait(false)) > 0)
-                    {
-                        await serverStream.WriteAsync(buffer, 0, bytesRead).ConfigureAwait(false);
-                    }
-                    serverStream.Flush();
+                    await clientStream.CopyToAsync(serverStream).ConfigureAwait(false);
                     serverSocket.Shutdown(SocketShutdown.Send);
                 }
-                catch (IOException)
+                catch (Exception ex)
                 {
-                    // Ignore rude disconnects from either side.
+                    HandleExceptions(ex);
                 }
             });
+
             Task serverCopyTask = Task.Run(async () =>
             {
                 try
                 {
-                    byte[] buffer = new byte[8000];
-                    int bytesRead;
-                    while ((bytesRead = await serverStream.ReadAsync(buffer, 0, buffer.Length).ConfigureAwait(false)) > 0)
-                    {
-                        await clientStream.WriteAsync(buffer, 0, bytesRead).ConfigureAwait(false);
-                    }
-                    clientStream.Flush();
+                    await serverStream.CopyToAsync(clientStream).ConfigureAwait(false);
+                    clientSocket.Shutdown(SocketShutdown.Send);
                 }
-                catch (IOException)
+                catch (Exception ex)
                 {
-                    // Ignore rude disconnects from either side.
+                    HandleExceptions(ex);
                 }
             });
 
-            await Task.WhenAny(new[] { clientCopyTask, serverCopyTask }).ConfigureAwait(false);
+            await Task.WhenAll(new[] { clientCopyTask, serverCopyTask }).ConfigureAwait(false);
+
+            /// <summary>Closes sockets to cause both tasks to end, and eats connection reset/aborted errors.</summary>
+            void HandleExceptions(Exception ex)
+            {
+                SocketError sockErr = (ex.InnerException as SocketException)?.SocketErrorCode ?? SocketError.Success;
+
+                // if aborted, the other task failed and is asking this task to end.
```

Comment:
nit: style.  We typically start sentences with capital letter. And we avoid extra white-space where possible. We don't use a white-space line between this comment and the code block that it refers to.

So, this would look like the below to conform to existing style of CoreFx networking code:

```c#
// If aborted, the other task failed and is asking this task to end.
if (sockErr == SocketError.OperationAborted)
{
    return;
}
```

Reply:
Same comment applies to other places in PR.